server/src/main/java/com/linecorp/centraldogma/server/metadata/MetadataService.java

```diff
@@ -133,7 +133,8 @@ public MetadataService(ProjectManager projectManager, CommandExecutor executor)
                 }
 
                 logger.warn("Adding missing repository metadata: {}/{}", projectName, repo);
-                final CompletableFuture<Revision> addRepoFuture = addRepo(Author.SYSTEM, projectName, repo);
+                final Author author = projectManager.get(projectName).repos().get(repo).author();
```

Comment:
Q) Doesn't `author()` call `history(..).join()` which is a blocking call?

Reply:
Seems like we use `CachingRepository` which caches the `author()` value. Feel free to ignore the above comment 😅

Reply:
If `CachingRepository` is applied, it doesn't do blocking call because it gets the author from the first commit. 
However, if the cache is not applied, yeah it does the blocking call.
Let me use the same approach in the `GitRepository`. Thanks!